pkg/artifactory/resource/security/resource_artifactory_group.go

```diff
@@ -5,257 +5,386 @@ import (
+				MarkdownDescription: "When this parameter is set, any new users defined in the system are automatically assigned to this group.",
+				Computed:            true,
+				Optional:            true,
+				Default:             booldefault.StaticBool(false),
+				Validators: []validator.Bool{
+					validatorfw.BoolConflict(true, path.Expressions{
+						path.MatchRelative().AtParent().AtName("admin_privileges"),
+					}...),
+				},
+			},
+			"admin_privileges": schema.BoolAttribute{
+				MarkdownDescription: "Any users added to this group will automatically be assigned with admin privileges in the system.",
+				Optional:            true,
+				Computed:            true,
+				Default:             booldefault.StaticBool(false),
+			},
+			"realm": schema.StringAttribute{
+				MarkdownDescription: "The realm for the group.",
+				Optional:            true,
+				Computed:            true,
+				Default:             stringdefault.StaticString("internal"),
+			},
+			"realm_attributes": schema.StringAttribute{
+				MarkdownDescription: "The realm attributes for the group.",
+				Optional:            true,
+			},
+			"users_names": schema.SetAttribute{
+				MarkdownDescription: "List of users assigned to the group. If not set or empty, Terraform will not manage group membership.",
+				ElementType:         types.StringType,
+				Optional:            true,
+			},
+			"detach_all_users": schema.BoolAttribute{
+				MarkdownDescription: "When this is set to `true`, an empty or missing usernames array will detach all users from the group.",
+				Optional:            true,
+				Computed:            true,
+				Default:             booldefault.StaticBool(false),
+			},
+			"watch_manager": schema.BoolAttribute{
+				MarkdownDescription: "When this override is set, User in the group can manage Xray Watches on any resource type. Default value is `false`.",
+				Optional:            true,
+				Computed:            true,
+				Default:             booldefault.StaticBool(false),
+			},
+			"policy_manager": schema.BoolAttribute{
+				MarkdownDescription: "When this override is set, User in the group can set Xray security and compliance policies. Default value is `false`.",
+				Optional:            true,
+				Computed:            true,
+				Default:             booldefault.StaticBool(false),
+			},
+			"reports_manager": schema.BoolAttribute{
+				MarkdownDescription: "When this override is set, User in the group can manage Xray Reports on any resource type. Default value is `false`.",
+				Optional:            true,
+				Computed:            true,
+				Default:             booldefault.StaticBool(false),
+			},
 		},
+	}
+}
 
-		Schema: GroupSchema,
+func (r *ArtifactoryGroupResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
 	}
+	r.client = req.ProviderData.(utilsdk.ProvderMetadata)
```

Comment:
Please rename this to `r.ProviderMetadata` or `r.ProviderData`.

Reply:
We probably don't need this func at all, as long as in CRUD func we get the client out from `req`.

```go
client := req.ProviderData.(utilsdk.ProviderMetadata).Client
```

Reply:
Renamed, removed, it does nothing.